### ISSUE TITLE
Wrong argument in LSTMDecoder's Adaptive Softmax Instantiation?

### ISSUE THREAD

**Issue by Alex-Mathai-98** [pytorch/fairseq#814]
`self.adaptive_softmax = AdaptiveSoftmax(num_embeddings, embed_dim, adaptive_softmax_cutoff,dropout=dropout_out)`

The following line exists in the `__init__` function of LSTMDecoder. After looking at the source code of AdaptiveSoftmax, the second argument "**embed_dim**" is supposed to be input size to the AdaptiveSoftmax matrix.

As the final LSTM layer outputs a vector of "**hidden_size**" shouldn't that be the input dimension to AdaptiveSoftmax ? I think that the above instantiation assumes that the word embedding size and the hidden size need to equal.

Please do correct me if I am wrong. Because only when I changed the second argument to "**hidden_size**" did the code stop throwing errors.

**Comment by myleott** [comment]
Good catch, can you submit a PR to fix it?

**Comment by Alex-Mathai-98** [comment]
Done. I've submitted a pull request.
